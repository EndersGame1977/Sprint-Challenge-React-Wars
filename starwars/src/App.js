import React, { useState, useEffect } from "react";
import axios from "axios";
import "./App.css";
import PeopleList from "../src/components/PeopleList";
import { Container, Header } from "semantic-ui-react";

const App = () => {
  const [peopleList, setPeopleList] = useState([]);

  useEffect(() => {
    const getPeopleList = () => {
      axios
        .get("https://swapi.co/api/people/")
        .then(res => {
          setPeopleList(res.data.results);
        })
        .catch(err => console.log(err));
    };
    getPeopleList();
  }, []);

  return (
    <Container
      textAlign="center"
      style={{ color: "#f9d71c", background: "black" }}
    >
      <Header as="h1" style={{ color: "#f9d71c" }}>
        React Wars
      </Header>
      <PeopleList peopleList={peopleList} />
    </Container>
  );
};
export default App;
