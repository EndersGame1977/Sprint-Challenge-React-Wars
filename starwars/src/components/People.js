import React from "react";
import { Segment } from "semantic-ui-react";

const PeopleList = props => {
  return (
    <Segment style={{ background: "black" }}>
      <h2>{props.people.name}</h2>
      <p>Eye Color: {props.people.eye_color}</p>
      <p>Hair Color: {props.people.hair_color}</p>
      <p>Skin Color: {props.people.skin_color}</p>
      <p>Birth Year: {props.people.birth_year}</p>
      <p>Gender: {props.people.gender}</p>
      <p>Height: {props.people.height}</p>
      <p>Mass: {props.people.mass}</p>
    </Segment>
  );
};

export default PeopleList;
