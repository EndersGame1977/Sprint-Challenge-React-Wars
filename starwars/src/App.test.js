import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import People from "../src/components/PeopleList";

import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("<App />", () => {
  it("Should Match SnapShot", () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("<People />", () => {
  it("Should Match SnapShot", () => {
    const tree = renderer.create(<People />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
