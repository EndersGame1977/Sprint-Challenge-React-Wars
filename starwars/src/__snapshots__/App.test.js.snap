// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should Match SnapShot 1`] = `
<div
  className="ui center aligned container"
  style={
    Object {
      "background": "black",
      "color": "#f9d71c",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "color": "#f9d71c",
      }
    }
  >
    React Wars
  </h1>
  <div
    className="ui container doubling stackable four column grid"
  />
</div>
`;
